packages/core/src/lib/utils.ts

```diff
@@ -10,14 +8,55 @@ import { readline } from './hook-engine.ts';
  * @return {string}
  */
 export function breakLines(content: string, width: number): string {
-  return content
-    .split('\n')
-    .flatMap((line) =>
-      wrapAnsi(line, width, { trim: false, hard: true })
-        .split('\n')
-        .map((str) => str.trimEnd()),
-    )
-    .join('\n');
+  const lines = content.split('\n');
+  const result: string[] = [];
+
+  for (const line of lines) {
+    let currentLine = '';
+    let visibleLength = 0;
+    let escapeSequence = '';
+    let inEscape = false;
+
+    for (let i = 0; i < line.length; i++) {
+      const char = line[i]!;
+
+      // Detect start of ANSI escape code
+      if (char === '\x1b') {
+        inEscape = true;
+        escapeSequence += char;
+        continue;
+      }
+
+      // If inside an escape sequence, accumulate it but don't count width
+      if (inEscape) {
+        escapeSequence += char;
+
+        if (/[a-zA-Z]/.test(char)) {
+          inEscape = false;
+          currentLine += escapeSequence;
+          escapeSequence = '';
+        }
```

Comment:
We should comment on the regex and explain how every escape sequence always ends with a single letter.